### PR TITLE
feat: implement a request id to trace requests

### LINKED ISSUE #586: Implement request IDs to trace errors
* allow consumers to pass a request ID in the request
* if none is provided, generate one
* include it in every log message of the request
* include it in error messages.

### PR DESCRIPTION
**Pull request by fengelniederhammer**
resolves #586

## PR Checklist
- [x] All necessary documentation has been adapted.
- [x] The implemented feature is covered by an appropriate end-to-end test.

### CODE REVIEW

File: lapis2/src/main/kotlin/org/genspectrum/lapis/logging/RequestId.kt [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import java.util.UUID
+
+@Component
+@RequestScope
+class RequestIdContext {
+    var requestId: String? = null
+}
+
+private const val REQUEST_ID_MDC_KEY = "RequestId"
+private const val HIGH_PRECEDENCE_BUT_LOW_ENOUGH_TO_HAVE_REQUEST_SCOPE_AVAILABLE = -100
```
Comment (JonasKellerer) [inline lapis2/src/main/kotlin/org/genspectrum/lapis/logging/RequestId.kt:21]: Is there docu we could link?

File: lapis2-docs/src/content/docs/concepts/request-id.mdx [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+If you report an error that occurred in LAPIS, please include the request id.
+This will greatly simplify identifying the problem in our log files.
+:::
+
+You can provide a request id in the request as an HTTP header `X-Request-Id`.
+This is useful if you want to correlate requests in your own log files with the LAPIS log files
+to track problems across systems.
+
+:::caution
+If you use the `X-Request-Id` header, make sure that the value is unique.
```
Comment (JonasKellerer) [inline lapis2-docs/src/content/docs/concepts/request-id.mdx:19]: What happens if it is not unique?

**Review by JonasKellerer — APPROVED** [review]
LGTM.
I only struggle with the scope. The request Id is for us very useful, but not for the enduser (which has no access to the LAPIS logs). Do we need to inform the user, that they can provide a request id? It seems they would not use that feature?

File: lapis2-docs/src/content/docs/concepts/request-id.mdx (hunk 2 above)
Comment (fengelniederhammer) [reply lapis2-docs/src/content/docs/concepts/request-id.mdx:19]: Well, we loose a bit of traceability...

File: lapis2/src/main/kotlin/org/genspectrum/lapis/logging/RequestId.kt (hunk 1 above)
Comment (fengelniederhammer) [reply lapis2/src/main/kotlin/org/genspectrum/lapis/logging/RequestId.kt:21]: Not really. This was try and error. `-500` doesn't work.

File: lapis2-docs/src/content/docs/concepts/request-id.mdx (hunk 2 above)
Comment (JonasKellerer) [reply lapis2-docs/src/content/docs/concepts/request-id.mdx:19]: Okay. Thats fine for me.

**Pull request merged**